services/placement/algo/algo_test.go

```diff
@@ -705,28 +709,30 @@ func TestAddInstance(t *testing.T) {
 	p := placement.NewPlacement(instances, ids, 1)
 
 	a := NewRackAwarePlacementAlgorithm(placement.NewOptions())
-	i2 := placement.NewEmptyInstance("i2", "r2", "", "endpoint", 1)
+	i2 := placement.NewEmptyInstance("i2", "r2", "", "e2", 1)
 	p, err := a.AddInstance(p, i2)
 	assert.NoError(t, err)
 	assert.Equal(t, 1, p.ReplicaFactor())
 	i1 = p.Instance("i1")
 	i2 = p.Instance("i2")
 	assert.Equal(t, 1, loadOnInstance(i1))
 	assert.Equal(t, 2, i1.Shards().NumShards())
+	assert.Equal(t, "e1", i1.Endpoint())
```

Comment:
Shouldn't we be checking the endpoint coming back from `AddInstance(..)`?  Otherwise we don't test that it got cloned properly.

e.g. `p.Instance(i1.InstanceID()).Endpoint()`?